File: TTK4255 - Robotsyn/Øving 6/solution/task1.py
import numpy as np
from normalize_points import *
from test_normalize_points import *

matches = np.loadtxt('../data/matches.txt')
pts = matches[:,:2]

print('Checking that the points satisfy the normalization criteria...')
pts_n,T = normalize_points(pts)
test_normalize_points(pts_n)

print('Checking that the transformation matrix performs the same operation...')
pts_n = T@np.column_stack((pts, np.ones(len(pts)))).T
pts_n = (pts_n[:2,:] / pts_n[2,:]).T
test_normalize_points(pts_n)

File: TTK4255 - Robotsyn/Øving 6/matlab/task1.m
clc; clf; clear;
matches = load('../data/matches.txt');
pts = matches(:,1:2);

[pts_n,T] = normalize_points(pts);

fprintf('Checking that the points satisfy the normalization criteria...\n');
test_normalize_points(pts_n);

fprintf('Checking that the transformation matrix performs the same operation...\n');
pts_n = (T*[pts' ; ones(1,size(pts,1))])';
pts_n = pts_n(:,1:2) ./ pts_n(:,3);
test_normalize_points(pts_n);

